src/charm_state.py

```diff
@@ -56,3 +58,20 @@ def from_charm(cls, charm) -> "CharmState":
         """
         hostname = charm.model.get_binding("website").network.bind_address
         return cls(hostname)
+
+    @classmethod
+    def check_rng_file(cls):
+        """Check if the rng-tools-debian file needs modification."""
+        file_modified = False
+        file = Path("/etc/default/rng-tools-debian")
+        if (
+            file.read_text(encoding="utf-8").count(
+                'RNGDOPTIONS="--fill-watermark=90% --feed-interval=1"'
+            )
+            > 1
+        ):
+            file_modified = True
+        if not file_modified:
+            file.write_text(
+                '\nRNGDOPTIONS="--fill-watermark=90% --feed-interval=1"', encoding="utf-8"
+            )
```

Comment:
I'm still not happy with this implementation.
I'd add a CharmState attribute that contains the content of the rng-tools-debian file, this content is loaded during CharmState instantiation
Then I'dd add a method to the PollenService that will set the CharmState attribute content if it's none and dump it to the file